li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -490,8 +490,17 @@ public long getRecordsSkipped() {
     return handledRecord;
   }
 
-  private byte[] parseAndMaybeTrackRecord(TopicPartition tp, long messageOffset, byte[] bytes) {
-    MessageAssembler.AssembleResult assembledResult = _messageAssembler.assemble(tp, messageOffset, bytes);
+  private byte[] parseAndMaybeTrackRecord(TopicPartition tp,
+                                          long messageOffset,
+                                          byte[] bytes,
+                                          Header header) {
+    MessageAssembler.AssembleResult assembledResult;
+    if (header == null) {
```

Comment:
This conditional doesn't cover all cases. You should call the old assembler API (the one without the `Header`) when:
1. `record.header()` == null OR
2. largeMessageHeader.version < V3

You can simplify this condition as:
```
if (header != null && header.lastHeader(LARGE_MESSAGE_HEAD).version == V3) { 

// Invoke the new assembler interface 

} else { 

// Invoke the old assembler interface

}

```

Reply:
You are correct that I should add version check for V3. I have put it in the implementation new assembler interface, which I think it is better place for us to call LargeMessageHeaderValue.fromBytes to get the LARGE_MESSAGE_HEAD instance. Let me know if it makes sense.